client/my-sites/backup/index.js

```diff
@@ -23,65 +26,77 @@ import WPCOMUpsellPage from 'my-sites/backup/wpcom-upsell';
 
 import { backupMainPath, backupActivityPath, backupRestorePath, backupDownloadPath } from './paths';
 
-export default function () {
-	if ( config.isEnabled( 'jetpack-cloud' ) || config.isEnabled( 'jetpack/features-section' ) ) {
-		/* handles /backup/activity, see `backupActivityPath` */
-		page( backupActivityPath(), siteSelection, sites, makeLayout, clientRender );
+const notFoundIfNotEnabled = ( context, next ) => {
+	const state = context.store.getState();
+	const siteId = getSelectedSiteId( state );
+	const showJetpackSection = isJetpackSectionEnabledForSite( state, siteId );
 
-		/* handles /backup/activity/:site, see `backupActivityPath` */
-		page(
-			backupActivityPath( ':site' ),
-			siteSelection,
-			navigation,
-			backupActivity,
-			wrapInSiteOffsetProvider,
-			showUpsellIfNoBackup,
-			wpcomUpsellController( WPCOMUpsellPage ),
-			makeLayout,
-			clientRender
-		);
+	if ( ! isJetpackCloud() && ! showJetpackSection ) {
+		return notFound( context, next );
+	}
 
-		/* handles /backup/:site/download/:rewindId, see `backupDownloadPath` */
-		page(
-			backupDownloadPath( ':site', ':rewindId' ),
-			siteSelection,
-			navigation,
-			backupDownload,
-			wrapInSiteOffsetProvider,
-			wpcomUpsellController( WPCOMUpsellPage ),
-			makeLayout,
-			clientRender
-		);
+	next();
+};
+
+export default function () {
+	/* handles /backup/activity, see `backupActivityPath` */
+	page( backupActivityPath(), siteSelection, sites, makeLayout, clientRender );
 
-		if ( config.isEnabled( 'jetpack/backups-restore' ) ) {
-			/* handles /backup/:site/restore/:rewindId, see `backupRestorePath` */
-			page(
-				backupRestorePath( ':site', ':rewindId' ),
-				siteSelection,
-				navigation,
-				backupRestore,
-				wrapInSiteOffsetProvider,
-				wpcomUpsellController( WPCOMUpsellPage ),
-				makeLayout,
-				clientRender
-			);
-		}
+	/* handles /backup/activity/:site, see `backupActivityPath` */
+	page(
+		backupActivityPath( ':site' ),
+		siteSelection,
+		navigation,
+		backupActivity,
+		wrapInSiteOffsetProvider,
+		showUpsellIfNoBackup,
+		wpcomUpsellController( WPCOMUpsellPage ),
+		notFoundIfNotEnabled,
```

Comment:
It doesn't change anything really, but I'm wondering if it'd make more sense to test for Jetpack sections before the upsell checks.

Reply:
The reason I put `notFoundIfNotEnabled` last is because it changes the value of `context.primary` -- but it just so happens, so do several of the middlewares that already exist 😅

Originally I dabbled with putting it first and "skipping" all the middlewares after it, but that puts a very tight coupling between `notFoundIfNotEnabled` and the last two middlewares, which are in charge of setting up the site layout and rendering it to the screen.

In the end, I decided it would be best for now to go with the existing pattern of only modifying `context.primary` and allowing the order of execution to work for us. It's also fortunate that we never actually render to the screen until `clientRender`, so I don't _believe_ we're wasting any significant CPU cycles with this approach. (If I'm wrong on this, someone please let me know!)

Reply:
Thanks for the detailed explanation! Makes perfect sense to me.